src/app/SubscriptionResumptionHelper.h

```diff
@@ -0,0 +1,53 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/AttributePathParams.h>
+#include <app/CASESessionManager.h>
+#include <app/SubscriptionResumptionStorage.h>
+
+namespace chip {
+namespace app {
+class InteractionModelEngine;
+class ReadHandler;
+
+class SubscriptionResumptionHelper : private SubscriptionResumptionStorage::SubscriptionInfo
+{
+public:
+    SubscriptionResumptionHelper();
+
+    ~SubscriptionResumptionHelper() {}
+
+private:
```

Comment:
if everything in this class is private. ... how does one use it? This does not look ok.

Reply:
Changed it